File: src/main/java/net/atos/client/zgw/shared/exception/ZgwErrorException.java
/*
 * SPDX-FileCopyrightText: 2021 Atos, 2025 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.client.zgw.shared.exception;

import java.net.URI;

import net.atos.client.zgw.shared.model.ZgwError;

/**
 * Exception thrown when an error occurred in the ZGW APIs.
 */
public class ZgwErrorException extends RuntimeException {

    private final ZgwError zgwError;

    public ZgwErrorException(final ZgwError zgwError) {
        this.zgwError = zgwError;
    }

    public ZgwError getZgwError() {
        return zgwError;
    }

    @Override
    public String getMessage() {
        return "%s [%d %s] %s (%s %s)"
                .formatted(
                        zgwError.getTitle(),
                        zgwError.getStatus(),
                        zgwError.getCode(),
                        zgwError.getDetail(),
                        uri(zgwError.getType()),
                        uri(zgwError.getInstance()));
    }

    private String uri(final URI uri) {
        return uri == null ? null : uri.toString();
    }
}

File: src/main/kotlin/nl/info/client/zgw/shared/exception/ZgwErrorException.kt
/*
 * SPDX-FileCopyrightText: 2021 Atos, 2025 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package nl.info.client.zgw.shared.exception

import nl.info.client.zgw.shared.model.ZgwError

/**
 * Exception thrown when an error occurred in the ZGW APIs.
 */
class ZgwErrorException(val zgwError: ZgwError) : RuntimeException() {

    override val message: String
        get() = "%s [%d %s] %s (%s %s)".format(
            zgwError.title,
            zgwError.status,
            zgwError.code,
            zgwError.detail,
            zgwError.type,
            zgwError.instance
        )
}
